scipy/misc/common.py

```diff
@@ -97,6 +97,7 @@ def derivative(func, x0, dx=1.0, n=1, args=(), order=3):
     Notes
     -----
     Decreasing the step size too small can result in round-off error.
+    Behavior at a non-differentiable point is unreliable.
```

Comment:
"Behavior" as opposed to "Result" because I don't think we want to imply that it will even return a result; it might raise an error.

Reply:
Yes, this simply makes the point.

Reply:
Nit: Calling it unreliable would confuse a user in me at least (consider it a datapoint). Maybe call it undefined? Since saying garbage in garbage out is probably not recommended in the official docs, maybe  "The function assumes that `func` is differentiable at `x0`. Otherwise the result is undefined" or some such.

Reply:
Much better. OK.

Reply:
@ev-br 

> Otherwise the result is undefined

would be wrong statement in this case, because the function does output a value rather than nan or something else. Saying the result is undefined doesn't sound right to me. Maybe another phrasing there ?

Reply:
I think we mean "we are not defining what the behavior should be".

Reply:
How about:
```
This function assumes that ``func`` is differentiable at ``x0``. 
If ``func`` is not differentiable at ``x0``, the behavior is undefined.
```

Reply:
I would rather that we explain and emphasize the nature of the central difference that's being computed and the necessity to specify an appropriate `dx` value. The problem isn't just _at_ the undifferentiable point but near it as well.

Reply:
@rkern Wouldn't that explanation make the point a bit complex or advanced ? However, I agree on your mark about unreliable results for near values. Also, I find "unreliable" more sound since it emphasize the unstability of the result, but I considered changing to inacurrate and use with may as it's a possibility to produce wrong results. In a simple manner, I'd suggest the two following statements:

```
1 - Behavior of this function may be inacurrate around ```x0``` and values near ```x0``` where the function ```func``` is not differentiable at ```x0```.

2 - Behavior of this function may be inacurrate where x ∈ (x0-e, x0+e) for some small e such that the function ```func``` is not differentiable at ```x0```.
```

What do you all think ?